🤩  🍱  Feat/theme send recv action +++

kcw-grunt opened the pull request:
## 📱 Description
<!-- Provide a brief description of what this PR accomplishes -->

## Platform
- [ ] Android
- [x] iOS
- [ ] Games-Unity
- [ ] DevOps (AWS)
- [ ] C/Golang 

## 🎯 Type of Change
<!-- Mark the relevant option with an [x] -->
- [ ] 🐛 Bug fix
- [x] ✨ New feature
- [ ] 🔧 Refactoring (code change that neither fixes a bug nor adds a feature)
- [ ] 📚 Documentation update
- [x] 🎨 UI/UX improvement
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement

## 📋 Changes
<!-- List the key changes made in this PR -->
This update make moderate steps to both add the design elements from @trk15 as well as connecting these views to the existing data sources. The key point is the **wallet is fully operational** but the access to those functionalities is not all complete.

The PR are will have explanations with the updates in the RC (Release Candidate) WIP ( Work in Progress) TBD (To be developed)

### RC Changes: Theoretically ready for production

### Tab Bar
 * Theme has been applied from design
 * Using SFSymbols matching close to the design 

| Tab Bar |
|--------------------|
|![tabbar](https://github.com/user-attachments/assets/66f26829-ce1d-441b-a004-c1e5c5c71bcc)|

### Balance Bento View
 * Theme has been applied from design
 * Re-added the toggle of Fiat and Litecoin animation
 * Using System Fonts (a whole overview of Fonts needs to be addressed)

| Balance Bento View |
|--------------------|
|![balancebento](https://github.com/user-attachments/assets/570a79e7-ea33-4ef2-9ed1-6dd47ba3102a)|

### Top Secret Favorites Bento View
 * Theme has been applied from design
 * Static design: Will stop development until after all features are complete

| Top Secret Favorites Bento View |
|--------------------|
|![topsecret](https://github.com/user-attachments/assets/5a547799-49e9-470b-b01d-217ab7ce2071)|

### LTC Fiat Ticker Bento View
 * Theme has been applied from design

| LTC Fiat Ticker Bento View |
|--------------------|
|![ticker](https://github.com/user-attachments/assets/fa29fa51-4c7a-4245-82ce-e4a4c5bd6496)|

kcw-grunt committed:
Complete layout of Bento Home views

Removed old TabBar and replaced with NewMainVC
Add in Bento Wireframes

kcw-grunt committed:
refactored

kcw-grunt committed:
Refactored build

kcw-grunt committed:
Remove cruft

kcw-grunt committed:
fixes minor bug in the theme mode

kcw-grunt committed:
WIP moving theme controls

kcw-grunt committed:
Merge branch 'bento' into feat/theme-send-recv-action

kcw-grunt committed:
WIP Upgrade to iOS 17 (17.6)

- Begin use the Metal Shader
- Added tiny notif animation...its no-op for UUserNotif
- Toolbar instead of TabBar
- WIP Building the transaction detail and Trans

kcw-grunt committed:
WIP : Polished the Balance Bento

- Polished the color scheme
- WIP: Added Transaction Detail
- Migrate the update from @trk15 designs
- Further polished main Layout
- worked in the Tutorial
- Polished Balance.

Polished the balance view

JosephSanjaya commented:
Thats look really cool!, can yous hare the background for the balance card, and moving background for home

kcw-grunt committed:
Added URL for the Notifications Settings

kcw-grunt commented:
> Thats look really cool!, can yous hare the background for the balance card, and moving background for home

Yes. But all of that is rendered. Its much more efficient @JosephSanjaya 

I will add notes for the colors in the code and tag you

kcw-grunt commented on brainwallet/Bento Effects /GridWaveContentView.swift:
```diff
@@ -0,0 +1,139 @@
+    @State
+    private var colorStyle: MainGradientStyle = .lightStyle
+
+    private let colorTriplet: (CIColor, CIColor, CIColor) =
```
These are the colors for the background @JosephSanjaya 
Im using this Shader that is a GPU processing feature so it is extremely efficient
Note I will change and polish this . Here is a great site: https://www.shadertoy.com/view/ctjfR1

I dont want to spend too much time until all the other features are operational

kcw-grunt commented on brainwallet/Bento Effects /GridWaveContentView.swift:
```diff
@@ -0,0 +1,139 @@
+            var startColor: CIColor
+            var endColor: CIColor
+
+            // Cycle through color pairs
```
This is a draft @JosephSanjaya . As long as the code is leveraging the GPU...Im less worried about matching. 
Here is a good example: https://proandroiddev.com/fancy-animated-ui-with-agsl-shaders-in-jetpack-compose-d962ba297bbb

The key is the shader language is common across iOS and Android.  So we should be able to easily port across.

kcw-grunt commented on brainwallet/Bento Effects /GridWaveContentView.swift:
```diff
@@ -0,0 +1,139 @@
+    }
+}
+
+// Draft design
```
Leaving as I want to test if there is more time.

kcw-grunt commented on brainwallet/Bento Effects /ThemeGradientHelpers.swift:
```diff
@@ -0,0 +1,167 @@
+    var colorThemeArray: [Color] {
+        switch self {
+        case .darkStyle:
+            return  [BrainwalletColor.surface.opacity(0.2), BrainwalletColor.grape.opacity(0.2), BrainwalletColor.midnight.opacity(0.1)]
```
As set of colors for the bentos

kcw-grunt commented on brainwallet/Bento Effects /ThemeGradientHelpers.swift:
```diff
@@ -0,0 +1,167 @@
+        }
+    }
+
+     var maskGradientStops: [Color] {
```
This should be platform common @JosephSanjaya . It sets the mask for the bento background animation. You should be able to copy almost directly. Or , if we like and update we both change it.  This is so the toolbar on the bottom is always the surface color theme

kcw-grunt commented on brainwallet/Bento Effects /ThemeGradientHelpers.swift:
```diff
@@ -0,0 +1,167 @@
+            .shadow(color: userPrefersDarkTheme ? Color.black.opacity(0.8) :
+                        Color(red: 0.341176, green: 0.329411, blue: 1).opacity(0.12),
+                    radius: 2, x: 1, y: 2.4)
+        /// Dark Mode : #000000 at 40% Opacity
```
Pulled the values from the Figma @JosephSanjaya . Again @trk15  has a big say on how the final look is. But we should be very close to matching for the shadows

kcw-grunt commented on brainwallet/Bento Effects /ThemeGradientHelpers.swift:
```diff
@@ -0,0 +1,167 @@
+    func body(content: Content) -> some View {
+        content
+            .foregroundStyle(userPrefersDarkTheme ? LinearGradient(colors: [Color(red: 0.5254901960784314,
+                                                           green: 0.4117647058823529,
```
This is the light and dark surface for the bento subviews.  We need to meet up with @trk15 on the final polish and match for both platforms @JosephSanjaya

kcw-grunt commented on brainwallet/Bento Effects /ThemeGradientHelpers.swift:
```diff
@@ -0,0 +1,167 @@
+    var userPrefersDarkTheme: Bool
+
+    var body: some View {
+        ZStack {
```
More values pulled from the Figma @JosephSanjaya .  Hope the hex values make it clear.

kcw-grunt commented on brainwallet/Bento Effects /ThemeGradientHelpers.swift:
```diff
@@ -0,0 +1,167 @@
+
+struct BalanceGameBackgroundView: View {
+
+    @Binding
```
Since the Balance and Game Bentos are different, using a different set of colors from the Figma @JosephSanjaya >
We can polish in 10 days

kcw-grunt commented on brainwallet/Bento Effects /aurora.metal:
```diff
@@ -0,0 +1,29 @@
+#include <metal_stdlib>
+using namespace metal;
+
+[[ stitchable ]] half4 aurora(float2 position, float time) {
```
WIP Aurora file

kcw-grunt commented on brainwallet/Bento Views/FavouritesBentoView.swift:
```diff
@@ -16,28 +16,104 @@ struct FavouritesBentoView: View {
+                    Spacer()
+                }
+
+                Group {
```
This is just holding space @JosephSanjaya . There is no plan to launch this section. But the design should match @JosephSanjaya

kcw-grunt commented on brainwallet/Info.plist:
```diff
@@ -111,6 +111,7 @@
 		<string>BarlowSemiCondensed-Light.ttf</string>
 		<string>BarlowSemiCondensed-Regular.ttf</string>
 		<string>BarlowSemiCondensed-Bold.ttf</string>
+		<string>BoldenVan.ttf</string>
```
The font from the Figma Font  ( BoldenVan) attached here : @JosephSanjaya 
[BoldenVan.zip](https://github.com/user-attachments/files/22991681/BoldenVan.zip)

kcw-grunt commented on brainwallet/Localizations/Localizable.xcstrings:
```diff
@@ -17613,6 +17616,9 @@
           }
         }
       }
+    },
```
Very early WIP

kcw-grunt commented on brainwallet/New Main Classes/NewMainView.swift:
The tab icons are from SFSymbols. Please see attached SVG files for your use @JosephSanjaya 
[SFSYMBOLS-BW.zip](https://github.com/user-attachments/files/22991738/SFSYMBOLS-BW.zip)

kcw-grunt committed:
Updated the GP Framework to accept iOS 17

kcw-grunt committed:
Further Polish

moved the iOS version to version 17
Moved all the colors to the Color file with naming
WIP: Work in the Transaction data source
Debugging the time delay of 3minutes in fetching the transactions
Added more func for the transaction detail

kcw-grunt commented:
Overlooking CI for now @josikie  @JosephSanjaya 
So tests wont pass.

josikie reviewed (approved):
LGTM. Thank you! @kcw-grunt
